### ISSUE TITLE
question about reproducing

### ISSUE THREAD

**Issue by XieJinluo** [mnansary/pytfBusterNet#1]
hello, did you try to reproduce the simi-branch as in the papers? I re-implemented the simi-branch of busternet in pyTorch and trained with the data provided, but i found that my f-score was far behind 0.45 as claimed in the papers.

**Comment by efkaf** [comment]
Same here. Any idea why?

**Comment by XieJinluo** [comment]
> Same here. Any idea why?

i am not sure. perhaps they actually didn't trained with the data they provided. or maybe... anyway, i already give up. what do you think?

**Comment by efkaf** [comment]
Not sure. I didn't use the pretrained weights of imagenet to train simi, as they mention. Did you? Besides this, whatever I tried did not drop the validation loss under 0.3, so I get way to many false positives.

**Comment by efkaf** [comment]
> > Same here. Any idea why?
> 
> i am not sure. perhaps they actually didn't trained with the data they provided. or maybe... anyway, i already give up. what do you think?

Not sure. I didn't use the pretrained weights of imagenet to train simi, as they mention. Did you? Besides this, whatever I tried did not drop the validation loss under 0.3, so I get way to many false positives.

**Comment by XieJinluo** [comment]
> > > Same here. Any idea why?
> > 
> > 
> > i am not sure. perhaps they actually didn't trained with the data they provided. or maybe... anyway, i already give up. what do you think?
> 
> Not sure. I didn't use the pretrained weights of imagenet to train it, as they mention. Did you? Besides that, whatever I tried did not drop the validation loss under 0.3, so I get way to many false positives.

Yes. I use the pretrained weights offered by torchvision. My validation loss was 0.17 and f-score was 0.6,. My training loss was 0.06 but f-score was only 0.3. Actually i found that there were some same object of copy-move in the training set and validation set, which makes the validation set not convincing.

**Comment by efkaf** [comment]
> > > > Same here. Any idea why?
> > > 
> > > 
> > > i am not sure. perhaps they actually didn't trained with the data they provided. or maybe... anyway, i already give up. what do you think?
> > 
> > 
> > Not sure. I didn't use the pretrained weights of imagenet to train it, as they mention. Did you? Besides that, whatever I tried did not drop the validation loss under 0.3, so I get way to many false positives.
> 
> Yes. I use the pretrained weights offered by torchvision. My validation loss was 0.17 and f-score was 0.6,. My training loss was 0.06 but f-score was only 0.3. Actually i found that there were some same object of copy-move in the training set and validation set, which makes the validation set not convincing.

So no luck even with imagenet weights, won't even try it. I don't know... but actually simi does not seem well trained even in BusterNet's predictions.

But I have one more question. Have you tried reproducing the results of CoMoFoD? The script includes only the analysis about the different kind of manipulations. Have you tried f-score with protocol B and discernibility analysis with the pretrained BusterNet? (these two functions are included only in the script for CASIA evaluation).

**Comment by XieJinluo** [comment]
I don't think this is about luck, and I didn't tried anything in CoMoFoD.

**Comment by efkaf** [comment]
> I don't think this is about luck, and I didn't tried anything in CoMoFoD.

I agree. The numbers do not match either.

**Comment by mnansary** [comment]
Hello, sorry for late reply. We did not reproduce as instructed in the paper. We merely took the idea of the arch and tried in on a very old dataset (micc-f2000) as mentioned in the README. Their dataset is probably around two hundred GiB in size and even with the use of tfrecord pipeline it will be very tedious for me.

**Comment by XieJinluo** [comment]
> Hello, sorry for late reply. We did not reproduce as instructed in the paper. We merely took the idea of the arch and tried in on a very old dataset (micc-f2000) as mentioned in the README. Their dataset is probably around two hundred GiB in size and even with the use of tfrecord pipeline it will be very tedious for me.

I can understand that. Thanks a lot for your reply！
